bitnami/jupyterhub/templates/hub/deployment.yaml

```
@@ -0,0 +1,227 @@
+apiVersion: {{ include "common.capabilities.deployment.apiVersion" . }}
+kind: Deployment
+metadata:
+  labels: {{- include "common.labels.standard" . | nindent 4 }}
+    app.kubernetes.io/component: hub
+    {{- if .Values.commonLabels }}
+    {{- include "common.tplvalues.render" ( dict "value" .Values.commonLabels "context" $ ) | nindent 4 }}
+    {{- end }}
+  name: {{ template "common.names.fullname" . }}-hub
+  {{- if .Values.commonAnnotations }}
+  annotations: {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
+  {{- end }}
+spec:
+  {{- if .Values.hub.updateStrategy }}
+  strategy: {{- toYaml .Values.hub.updateStrategy | nindent 4 }}
+  {{- end }}
+  selector:
+    matchLabels: {{- include "common.labels.matchLabels" . | nindent 6 }}
+      app.kubernetes.io/component: hub
+  template:
+    metadata:
+      {{- if .Values.hub.podAnnotations }}
+      annotations: {{- include "common.tplvalues.render" (dict "value" .Values.hub.podAnnotations "context" $) | nindent 8 }}
+      {{- end }}
```

Add the checksums of the configmap/secret when these are created:
```suggestion
      annotations:
        {{- if .Values.hub.podAnnotations }}
        {{- include "common.tplvalues.render" ( dict "value" .Values.hub.podAnnotations "context" $ ) | nindent 4 }}
        {{- end }}
        {{- if not .Values.hub.existingConfigmap }}
        checksum/hub-config: {{ include (print $.Template.BasePath "/hub/configmap.yaml") . | sha256sum }}
        {{- end }}
        {{- if not .Values.hub.existingSecret }}
        checksum/hub-secret: {{ include (print $.Template.BasePath "/hub/secret.yaml") . | sha256sum }}
        {{- end }}
```

---

Done